### PR TITLE
[KM29] Add Amazon Elastic Block Store (EBS) CSI driver to cluster

### PR DESCRIPTION
**Pull request by paulbes**
Add [Amazon Elastic Block Store (EBS) CSI driver](https://github.com/kubernetes-sigs/aws-ebs-csi-driver) to the Kubernetes cluster

Storage class declaration:
```
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: ebs-sc
provisioner: ebs.csi.aws.com
volumeBindingMode: WaitForFirstConsumer
```

Persistent volume claim:
```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: block-claim
spec:
  accessModes:
    - ReadWriteOnce
  volumeMode: Block
  storageClassName: ebs-sc
  resources:
    requests:
      storage: 10Gi
```

Results in the following EBS volume:
image

## Description
This makes it possible to create `PersistentVolumeClaims` and attach them to pods.

**NB:** This does not work with pods scheduled onto fargate nodes

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I have updated the release notes (for the next release).

### CODE REVIEW

File: pkg/apis/okctl.io/v1alpha1/cluster_v1alpha1.go [hunk 1]
```diff
@@ -196,6 +196,11 @@ type ClusterIntegrations struct {
 	// +optional
 	Autoscaler bool `json:"autoscaler,omitempty"`
 
+	// Blockstorage if set to true will install the EBS CSI block storage driver into the
+	// cluster, which makes it possible to create PersistentVolumeClaims in AWS
+	// +optional
```
Comment (deifyed) [inline pkg/apis/okctl.io/v1alpha1/cluster_v1alpha1.go:201]: What does +optional actually do?

**Review by deifyed — APPROVED** [review]
Remember to update the changelog

**Review by yngvark — APPROVED** [review]
Looks good!

**Comment by yngvark** [comment]
Does the user have to specify 

```yaml
  storageClassName: ebs-sc
```

in their PVC definition?

If yes - can we set it to default, so the user doesn't have to care about the underlying storage driver?

File: pkg/apis/okctl.io/v1alpha1/cluster_v1alpha1.go (hunk 1 above)
Comment (paulbes) [reply pkg/apis/okctl.io/v1alpha1/cluster_v1alpha1.go:201]: It is based on k8s api machinery (if I remember correctly), essentially we could use it to autogenerate JSON schema, validation, etc.

**Comment by paulbes** [comment]
Nice catch, yes we do, and yes we should make some or another storageclass a default. 👍🏽

**Review by deifyed — APPROVED** [review]

**Review by yngvark — APPROVED** [review]

**Pull request merged**